sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1883,7 +1883,19 @@ class Analyzer(
       // Second, we group extractedWindowExprBuffer based on their Partition and Order Specs.
       val groupedWindowExpressions = extractedWindowExprBuffer.groupBy { expr =>
         val distinctWindowSpec = expr.collect {
-          case window: WindowExpression => window.windowSpec
+          case window: WindowExpression =>
+            val winExpr = window.windowFunction
+            val distinctOpt = winExpr.find (expr => expr.isInstanceOf[AggregateExpression]
+                && expr.asInstanceOf[AggregateExpression].isDistinct)
+            if (distinctOpt.nonEmpty && window.windowSpec.orderSpec.nonEmpty) {
+              failAnalysis(s"ORDER BY cannot be used with DISTINCT: $window")
```

Comment:
Just out of curiosity, does hive have the same limitation? If so, the current way, roughly ordered rows and checking previous row for distinct windows makes sense to me.

Reply:
Yes, it's supported and some other engines like ODPS...
Thanks a lot for comment @HyukjinKwon

Reply:
Was wondering if they have the same limitation for clarification.